### DISCUSSION TITLE
Should the `full_url` validaiton rule check for TLD?

### DISCUSSION THREAD

**Discussion by vendion** [goravel/goravel#532] [Ideas]
Currently, the `full_url` validation rule checks for the existence of the schema (http://, https://, etc...) and then seems to accept anything after that. That means with the current implantation, input like "http://foo" passes (which is fine in the context of an intranet site or a machine's hostname, etc…) but I doubt that should be considered a "full" URL.

Maybe the current `full_url` could be renamed to `url` and a new `full_url` rule is defined that checks for the existing of a possible TLD. It doesn't need to be too strict, otherwise could get complicated as new TLDs get added/approved fairly often, plus could be an issue with subdomains and such.

**Comment by hwbrzzl** [comment]
Good catch, basically, it's a function of [the third package](https://github.com/gookit/validate), we don't modify it specially. If there is no strong requirement, we can keep it for now. Welcome a deeper discussion.

<img width="1129" alt="image" src="https://github.com/user-attachments/assets/cb9570a0-bd4f-46f6-8923-6162378033e8">

**Reply by vendion** [reply]
Okay, it looks like there is already a ticket for this in the upstream repo https://github.com/gookit/validate/issues/138 and seems that their `url`/`fullUrl` regexes are completely broken currently.

**Reply by hwbrzzl** [reply]
Yes, good catch.
